lib/livebook/intellisense.ex

```diff
@@ -207,7 +221,96 @@ defmodule Livebook.Intellisense do
       insert_text: name
     }
 
-  @ordered_kinds [:field, :variable, :module, :struct, :interface, :function, :type]
+  defp keyword_macro?(name) do
+    def? = name |> Atom.to_string() |> String.starts_with?("def")
+
+    def? or
+      name in [
+        :alias,
+        :import,
+        :require,
+        :use,
+        :case,
+        :cond,
+        :for,
+        :if,
+        :quote,
+        :raise,
+        :receive,
+        :throw,
+        :try,
+        :unless,
+        :with
+      ]
+  end
+
+  defp env_macro?(name) do
+    name in [:__ENV__, :__MODULE__, :__DIR__, :__STACKTRACE__, :__CALLER__]
+  end
+
+  defp extra_completion_items(hint) do
+    items = [
+      %{
+        label: "do",
+        kind: :keyword,
+        detail: "do-end block",
+        documentation: nil,
+        insert_text: "do\n  $0\nend"
+      },
+      %{
+        label: "end",
+        kind: :keyword,
+        detail: "do-end block",
+        documentation: nil,
+        insert_text: "end"
+      },
+      %{
+        label: "fn",
+        kind: :keyword,
+        detail: "anonymous function",
+        documentation: nil,
+        insert_text: "fn"
+      },
+      %{
+        label: "true",
+        kind: :keyword,
+        detail: "boolean",
+        documentation: nil,
+        insert_text: "true"
+      },
+      %{
+        label: "false",
+        kind: :keyword,
+        detail: "boolean",
+        documentation: nil,
+        insert_text: "false"
+      },
+      %{
+        label: "nil",
+        kind: :keyword,
+        detail: "special atom",
+        documentation: nil,
+        insert_text: "nil"
+      },
+      %{
+        label: "when",
+        kind: :keyword,
+        detail: "guard operator",
+        documentation: nil,
+        insert_text: "when"
+      }
```

Comment:
I didn't include `else`, `after`, etc because it breaks automatic indentation, example:

```elixir
if true do
  els<CURSOR>
end
```

Typing `else` automatically indents as expected, but accepting a completion item doesn't.

I think it's fine because for those keywords there are no other completions to intercept enter.

Reply:
Can't we have the same issue with `end`? 🤔

Reply:
Good catch, most of the time `end` is inserted automatically, so I missed that :D